SocialNetworkApi/Controllers/UserController.cs

```diff
@@ -26,42 +26,42 @@ public async Task<IActionResult> CreateUser([FromBody] CreateUserRequest request
         );
     }
 
-        [HttpGet]
-        [Route("users/{userId:guid}")]
-        public async Task<IActionResult> GetUserById(Guid userId)
-        {
-            var result = await _userService.GetByIdAsync(userId);
-            return result.Data is null
-                ? Problem(statusCode: StatusCodes.Status404NotFound, detail: $"User not found (userId {userId})")
-                : Ok(result.Data);
-        }
+    [HttpGet]
+    [Route("{userId:guid}")]
+    public async Task<IActionResult> GetUserById(Guid userId)
+    {
+        var result = await _userService.GetByIdAsync(userId);
+        return result.Data is null
+            ? Problem(statusCode: StatusCodes.Status404NotFound, detail: $"User not found (userId {userId})")
+            : Ok(result.Data);
+    }
 
-        [HttpPut]
-        [Route("users/{userId:guid}")]
-        public async Task<IActionResult> Edit([FromRoute] Guid userId, [FromBody] EditUserRequest request)
-        {
-            var user = request.ToDomain();
-            var result = await _userService.UpdateAsync(userId, user);
+    [HttpPut]
+    [Route("{userId:guid}")]
+    public async Task<IActionResult> Edit([FromRoute] Guid userId, [FromBody] UpdateUserResponse request)
+    {
+        var user = request.ToDomain();
+        var result = await _userService.UpdateAsync(userId, user);
 
         return result.Success
             ? Ok(result.Data)
             : Problem(statusCode: StatusCodes.Status400BadRequest, detail: "Failed to update user");
     }
 
-        [HttpDelete]
-        [Route("users/{userId:guid}")]
-        public async Task<IActionResult> DeleteUser(Guid userId)
-        {
-            await _userService.DeleteAsync(userId);
-            return NoContent();
-        }
+    [HttpDelete]
+    [Route("{userId:guid}")]
+    public async Task<IActionResult> DeleteUser(Guid userId)
+    {
+        await _userService.DeleteAsync(userId);
+        return NoContent();
+    }
 
-        [HttpGet]
-        [Route("users")]
-        public async Task<IActionResult> GetAllUsers()
-        {
-            var users = await _userService.GetAllAsync(); 
-            var userResponses = users.Select(UserResponse.FromDomain); 
+    [HttpGet]
+    [Route("allusers")]
+    public async Task<IActionResult> GetAllUsers()
+    {
+        var users = await _userService.GetAllAsync();
+        var userResponses = users.Select(UserResponse.FromDomain);
 
         return Ok(userResponses);
     }
```

Comment:
The name allusers could be improved to users, getAllUsers, or listUsers to make it more descriptive and in line with REST standards.